app_dart/test/request_handlers/check_for_waiting_pull_requests_test.dart

```diff
@@ -0,0 +1,402 @@
+import '../src/request_handling/fake_http.dart';
+import '../src/request_handling/fake_logging.dart';
+
+const String base64LabelId = 'base_64_label_id';
+
+void main() {
+  group('check for waiting pull requests', () {
+    CheckForWaitingPullRequests handler;
+
+    FakeHttpRequest request;
+    FakeConfig config;
+    FakeClientContext clientContext;
+    FakeAuthenticationProvider auth;
+    FakeLogging log;
+    FakeGraphQLClient githubGraphQLClient;
+
+    ApiRequestHandlerTester tester;
+
+    final List<PullRequestHelper> flutterRepoPRs = <PullRequestHelper>[];
+    final List<PullRequestHelper> engineRepoPRs = <PullRequestHelper>[];
+
+    setUp(() {
+      request = FakeHttpRequest();
+      config = FakeConfig();
+      clientContext = FakeClientContext();
+      auth = FakeAuthenticationProvider(clientContext: clientContext);
+      log = FakeLogging();
+      githubGraphQLClient = FakeGraphQLClient();
+      flutterRepoPRs.clear();
+      engineRepoPRs.clear();
+      PullRequestHelper._counter = 0;
+
+      githubGraphQLClient._mutateResultForOptions =
+          (MutationOptions options) => QueryResult();
+
+      githubGraphQLClient._queryResultForOptions = (QueryOptions options) {
+        expect(options.variables['sOwner'], 'flutter');
+        expect(options.variables['sLabelName'],
+            config.waitingForTreeToGoGreenLabelNameValue);
+
+        final String repoName = options.variables['sName'];
+        if (repoName == 'flutter') {
+          return createQueryResult(flutterRepoPRs);
+        } else if (repoName == 'engine') {
+          return createQueryResult(engineRepoPRs);
+        } else {
+          fail('unexpected repo $repoName');
+        }
+      };
+
+      tester = ApiRequestHandlerTester(request: request);
+      config.waitingForTreeToGoGreenLabelNameValue =
+          'waiting for tree to go green';
+      config.githubGraphQLClient = githubGraphQLClient;
+
+      handler = CheckForWaitingPullRequests(
+        config,
+        auth,
+        loggingProvider: () => log,
+      );
+    });
+
+    void _verifyQueries() {
+      githubGraphQLClient.verifyQueries(
+        <QueryOptions>[
+          QueryOptions(
+            document: labeledPullRequestsWithReviewsQuery,
+            fetchPolicy: FetchPolicy.noCache,
+            variables: <String, dynamic>{
+              'sOwner': 'flutter',
+              'sName': 'flutter',
+              'sLabelName': config.waitingForTreeToGoGreenLabelNameValue,
+            },
+          ),
+          QueryOptions(
+            document: labeledPullRequestsWithReviewsQuery,
+            fetchPolicy: FetchPolicy.noCache,
+            variables: <String, dynamic>{
+              'sOwner': 'flutter',
+              'sName': 'engine',
+              'sLabelName': config.waitingForTreeToGoGreenLabelNameValue,
+            },
+          ),
+        ],
+      );
+    }
+
+    test('Merges first PR in list, all successful', () async {
```

Comment:
Can we also add a test with a lot of pull requests (e.g., 200) and make sure that we won't crash (because we set some limit to 100 in our code)?

Reply:
The API has a max of 100 and we specify we only want 100 in our query. Do you still think such a test is useful?

Reply:
Ok, if the limit is on the GIthub side, maybe it's not very useful to test on our side.